### PR TITLE
Add ability to skip ssl check

### PR DESCRIPTION
**Pull request by F1ashhimself**

### CODE REVIEW

**Comment by F1ashhimself** [comment]
@olivergondza could you please take a quick look at latest failed test?

**Comment by F1ashhimself** [comment]
@olivergondza thank you! This feature is present in JClouds plugin and such plugins as Stash e.t.c. In our case we have local openstack (only intranet) and local jenkins but openstack have self-signed cert... so we should be able to skip this check :(

**Comment by olivergondza** [comment]
Well, an alternative would be to truck the cert in JVM truststore - even though it is self signed.

**Comment by F1ashhimself** [comment]
Anyway it is up to you to merge or not to merge :)

All tests passed.

File: src/main/java/jenkins/plugins/openstack/compute/JCloudsCloud.java [hunk 1]
```diff
@@ -68,6 +68,7 @@
     private String credentialId;
 
     public final @Nonnull String endPointUrl;
+    public final boolean ignoreSsl;
```
Comment (olivergondza) [inline src/main/java/jenkins/plugins/openstack/compute/JCloudsCloud.java]: Please make all new fields private.

File: src/main/java/jenkins/plugins/openstack/compute/slaveopts/BootSource.java [hunk 2]
```diff
@@ -272,21 +272,23 @@ public int hashCode() {
...
-                return makeListBoxModelOfAllNames(name, endPointUrl,credentialId, zone);
+                return makeListBoxModelOfAllNames(name, endPointUrl, ignoreSsl, credentialId, zone);
             }
 
             @Restricted(DoNotUse.class)
             public FormValidation doCheckName(@QueryParameter String value,
                     // authentication fields can be in two places relative to us.
                                               @RelativePath("../..") @QueryParameter("endPointUrl") String endPointUrlCloud,
                                               @RelativePath("../../..") @QueryParameter("endPointUrl") String endPointUrlTemplate,
+                                              @QueryParameter boolean ignoreSsl,
```
Comment (olivergondza) [inline src/main/java/jenkins/plugins/openstack/compute/slaveopts/BootSource.java]: I am afraid we need this to be duplicated as all the other cloud fields in the signature for it to work.

File: src/main/java/jenkins/plugins/openstack/compute/slaveopts/BootSource.java [hunk 3]
```diff
@@ -391,11 +393,12 @@ public FormValidation doCheckName(@QueryParameter String value,
                     // us.
                                               @RelativePath("../..") @QueryParameter("endPointUrl") String endPointUrlCloud,
                                               @RelativePath("../../..") @QueryParameter("endPointUrl") String endPointUrlTemplate,
+                                              @QueryParameter boolean ignoreSsl,
```
Comment (olivergondza) [inline src/main/java/jenkins/plugins/openstack/compute/slaveopts/BootSource.java]: Ditto.

File: src/test/java/jenkins/plugins/openstack/compute/SlaveOptionsDescriptorTest.java [hunk 4]
```diff
@@ -247,14 +248,15 @@ public void doCheckAvailabilityZoneGivenNoAZAndMultipleZoneToChooseFromThenRetur
...
-        final FormValidation actual = d.doCheckAvailabilityZone(value, def, "OSurl", "OSurl",openstackAuth, openstackAuth, "OSzone", "OSzone");
+        final FormValidation actual = d.doCheckAvailabilityZone(value, def, "OSurl", false, "OSurl",openstackAuth, openstackAuth, "OSzone", "OSzone");
 
         assertThat(actual, hasState(expected));
     }
 
     @Test
     public void fillDependencies() throws Exception {
         List<String> expected = Arrays.asList(
+                "ignoreSsl",
```
Comment (olivergondza) [inline src/test/java/jenkins/plugins/openstack/compute/SlaveOptionsDescriptorTest.java]: Ditto.

File: src/test/java/jenkins/plugins/openstack/compute/JCloudsCloudTest.java [hunk 5]
```diff
@@ -212,7 +212,7 @@ public void testConfigRoundtrip() throws Exception {
 
         String beans = "credentialId,endPointUrl,zone";
```
Comment (olivergondza) [inline src/test/java/jenkins/plugins/openstack/compute/JCloudsCloudTest.java]: The new bean field needs to be added here to be tested.

**Comment by F1ashhimself** [comment]
Sure, I will look into this tommorow. Thank you!

**Comment by F1ashhimself** [comment]
@olivergondza I have tried to make jenkins/plugins/openstack/compute/JCloudsCloud.java:71 this private as you said but in this case this (jenkins.plugins.openstack.compute.JCloudsCloudTest#testConfigRoundtrip) test will fail:
```
 JCloudsCloudTest.testConfigRoundtrip:223 No such property ignoreSsl on class jenkins.plugins.openstack.compute.JCloudsCloud
```
all other comments were fixed.

**Comment by olivergondza** [comment]
@F1ashhimself, yes, you have to provide a getter with it (though the setter is undesirable) for several things to work correctly. Propagating the value to webform is another one.

**Comment by F1ashhimself** [comment]
Done.

**Pull request merged**